woocommerce/woocommerce
# Allow order summary headings to be customized via entities

### Changes proposed in this Pull Request:

Closes #52043

This feature allows the editing of various headings within the cart and checkout order summary pages. Where the headings share the same purpose we save them using entities so they can be edited once and updated in both blocks.

There are some headings I've excluded:

* Taxes. Taxes by default in the editor displays taxes as line items and the way we display them differs from non line item taxes. These are displayed with their names in a list. So we can't easily edit the generic Tax heading and even if we did it might be confusing to merchants to have a discrepancy between f/e and b/e. We could enhance this somehow at some point but for now I think it's better to leave it not editable

Also note: I have stored the "Order Summary" heading in an entity even though right now it's not shared across cart and checkout, (cart uses "content" to store it's own header right now and they're different). I thought it could be useful to do this now though and enhance both to match in future because it (could) make sense to consolidate them, but i'm open to either removing that from this PR, or changing that one to be straight update/save of a regular block attribute.




### How to test the changes in this Pull Request:

For the following headings:

* "Subtotal" heading in cart and checkout order summary block
* "Total" at bottom of cart and checkout order summary block
* "Add a coupon" heading in cart and checkout order summary block
* "Fee" heading in cart and checkout order summary block
* "Delivery" heading in cart and checkout order summary block

Test:

* Edit the heading in either cart or checkout page
* Save, note that the entity editing confirmation should be shown, note that the label for the heading you're editing makes sense
* Refresh the editor, confirm that your new text shows in the place of that heading
* Go to editor for the other page (e.g. if you were editing checkout, go to editor for cart page) and ensure your new heading text also displays there in the correct place
* Go to frontend page for cart and checkout, confirm that your new text is displayed in the right heading

Example:


https://github.com/user-attachments/assets/59a5767f-e0ce-463e-abb4-65feb606bbe9




Also confirm that before editing headings that they all display correctly.

Note for testing **fee** heading you'll need to add a mock fee so that the heading displays on front-end.
Here's some code you can run to ensure that a fee is added:

```
/**
 * Add a standard $ value surcharge to all transactions in cart / checkout
 */
add_action( 'woocommerce_cart_calculate_fees','wc_add_surcharge' ); 
function wc_add_surcharge() { 
    global $woocommerce; 

    if ( is_admin() && ! defined( 'DOING_AJAX' ) ) 
        return;
    $fee = 1.00;
    $woocommerce->cart->add_fee( 'Surcharge', $fee, true, 'standard' );  
}
```





### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 
Allow headings in cart/checkout order summary blocks to be edited

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
